seed/src/event.rs

```diff
@@ -0,0 +1,61 @@
+use std::net::SocketAddr;
+
+use librad::{meta::project::ProjectInfo, net::peer::PeerApi, peer::PeerId, uri::RadUrn};
+
+use crate::{guess_user, Error, Project, Signer};
+
+/// An event generated by the seed node.
+#[derive(Debug, Clone)]
+pub enum Event {
+    /// The seed node is listening for peer connections.
+    Listening(SocketAddr),
+    /// A peer has connected.
+    PeerConnected {
+        peer_id: PeerId,
+        urn: Option<RadUrn>,
+        name: Option<String>,
+    },
+    /// A peer has disconnected.
+    PeerDisconnected(PeerId),
+    /// A project has been tracked from a peer.
+    ProjectTracked(Project, PeerId),
+}
+
+impl Event {
+    pub(crate) async fn peer_connected(
```

Comment:
What's the reason for the excessive use of crate visibility? Why not instead control the re-exports?

Reply:
Well the thing is I need to export `Event`, but having this as simply `pub` will allow users to call `Event::peer_connected` afaict.